### ISSUE TITLE
[PATCH] Distrib throws exception when accessing /users/username/ where username exists but has no public_html

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11510]
|<img alt="Jerub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Jerub reported|
|-|-|
|Trac ID|trac#551|
|Type|defect|
|Created|2004-03-15 09:32:57Z|

Attachments:

* [distrib-publichtml.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/a4c/a4c36ded9dbf60a5a9dceb0d0a1e3b17ac92f8cf/d8476841bf050f4c10899daa00e3e69c007be4b0.diff) (585 bytes) - added by Jerub on 2004-03-15 09:32:57Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__551 551
type__defect defect
reporter__Jerub Jerub
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1079343177000000 1079343177000000
changetime__1142787876000000 1142787876000000
version__ 
owner__ 
cc__radix cc__Jerub
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Jerub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Jerub commented|
|-|-|

```
#!html
<pre>
On a system where 'joeblogs' is a user, but '/home/joeblogs/public_html/'
doesn't exist, UserDistrib throws an exception when
http://server/distrib/users/joeblogs/ is accessed, because it returns a
static.File(path) where path doesn't exist.

Attached is a diff which will check that the directory exists before writing to
it. The patch is tested.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph commented|
|-|-|

```
#!html
<pre>
I haven't assigned anything to you in a while.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix set status to `closed`|
|-|-|

(In [16358]) Don't raise an exception when a user's public_html doesn't exist. Fixes [#11510](https://github.com/twisted/twisted/issues/11510) (!).
Author: Jerub
Reviewer: radix

The unfortunate lack of tests is due to untestable design on the part of 
UserDirectory. Use TDD next time.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
